backend/tracim_backend/views/core_api/user_controller.py

```diff
@@ -185,7 +187,29 @@ def user_workspace(self, context, request: TracimRequest, hapic_data=None):
             include_with_role=hapic_data.query.show_workspace_with_role,
             parents_ids=hapic_data.query.parent_ids,
         )
-        return [wapi.get_workspace_with_context(workspace) for workspace in workspaces]
+        return [
+            wapi.get_workspace_with_context(workspace, user=request.candidate_user)
+            for workspace in workspaces
+        ]
+
+    @hapic.with_api_doc(tags=[SWAGGER_TAG__USER_CONTENT_ENDPOINTS])
+    @check_right(has_personal_access)
+    @hapic.input_path(UserIdPathSchema())
+    @hapic.input_query(UserWorkspaceFilterQuerySchema())
+    @hapic.output_body(WorkspaceMemberSchema(many=True))
+    def user_role_workspace(self, context, request: TracimRequest, hapic_data=None):
+        """
+        Get list of user role in workspace
```

Comment:
Get list of user role in workspace -> Get list of user roles in the given workspace

Reply:
It's indeed unclear, this function return the list of all the role of the given user in each workspace he is in;
Get list of user role in workspace -> Get list of all roles of the given user

Reply:
fixed